[49581] Translate names from common files when seeding

cbliard opened the pull request:
https://community.openproject.org/wp/49581

So that the role names, the color names, and the document categories are correctly translated when another language than English is selected.

cbliard committed:
[49581] Translate names from common files when seeding

aaron-contreras reviewed (commented):
Ah! Very nice 🖌️

Just a question to confirm my understanding of the fix:

So only `standard.yml` and `bim.yml` were being picked up for translation since those were the arguments to `seed_name` `Source::SeedDataLoader` but not `common.yml`, however now when using `seed_file.name` and translating each file individually we're able to search for the `common` key under the translation files as well?

cbliard commented:
> So only `standard.yml` and `bim.yml` were being picked up for translation since those were the arguments to `seed_name` `Source::SeedDataLoader` but not `common.yml`, however now when using `seed_file.name` and translating each file individually we're able to search for the `common` key under the translation files as well?

Not exactly. Before the fix, both `standard.yml` and `common.yml` files [were picked](https://github.com/opf/openproject/blob/0a7a93b2b27e56ee04e22389151f138b2cfa4900/app/seeders/source/seed_data_loader.rb#L67), and they were then merged together in a single hash, and then translated using `seeds.#{seed_name}` as i18n prefix. This works for strings coming from `standard.yml`, but not for the ones coming for `common.yml` as the prefix should be `seeds.common`, but the information about the originating file is lost as the hashes have been merged together

The fix is to do things in the right order: get `standard.yml` and `common.yml`, translate them using the appropriate i18n prefix, and then merge them.

I'll have a look on why the tests are failing.

cbliard committed:
Remove db string length constraint on some fields

There is a mismatch between some database fields and model attributes
about the max length. This leads to a model being valid from a
ActiveRecord point of view, but failing to be inserted in database.

Such invalid constraints have been removed from the database, and the
remaining ones are controlled by a unit test.

This was also causing seeding issues for some languages as translated
names are longer than allowed.

Fixes https://community.openproject.org/wp/49583

cbliard commented:
Tests are failing because of https://community.openproject.org/wp/49583. The last commit fixes it.
